packages/infra/lib/quest/quest-stack.ts

```diff
@@ -0,0 +1,441 @@
+import { Duration, NestedStack, NestedStackProps } from "aws-cdk-lib";
+import { SnsAction } from "aws-cdk-lib/aws-cloudwatch-actions";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { Function as Lambda } from "aws-cdk-lib/aws-lambda";
+import { SqsEventSource } from "aws-cdk-lib/aws-lambda-event-sources";
+import * as s3 from "aws-cdk-lib/aws-s3";
+import * as secret from "aws-cdk-lib/aws-secretsmanager";
+import { Queue } from "aws-cdk-lib/aws-sqs";
+import { Construct } from "constructs";
+import { EnvConfig } from "../../config/env-config";
+import { EnvType } from "../env-type";
+import { createLambda } from "../shared/lambda";
+import { LambdaLayers } from "../shared/lambda-layers";
+import { buildSecret } from "../shared/secrets";
+import { LambdaSettings, QueueAndLambdaSettings } from "../shared/settings";
+import { createQueue } from "../shared/sqs";
+import { QuestAssets } from "./types";
+
+const sftpActionTimeout = Duration.seconds(30);
+const sendPatientRequestLambdaTimeout = Duration.seconds(30);
+const sendBatchRequestLambdaTimeout = Duration.minutes(5);
+const receiveResponseLambdaTimeout = Duration.seconds(30);
+const convertPatientResponseLambdaTimeout = Duration.seconds(30);
+const convertBatchResponseLambdaTimeout = Duration.minutes(5);
+const alarmMaxAgeOfOldestMessage = Duration.hours(1);
+
+interface QuestSettings {
+  sendPatientRequest: QueueAndLambdaSettings;
+  sendBatchRequest: QueueAndLambdaSettings;
+  receiveResponse: QueueAndLambdaSettings;
+}
+
+interface QuestLambdaSettings {
+  sftpAction: LambdaSettings;
+  convertPatientResponse: LambdaSettings;
+  convertBatchResponse: LambdaSettings;
+}
+
+const settings: QuestSettings = {
+  sendPatientRequest: {
+    name: "QuestSendPatientRequest",
+    entry: "quest/send-patient-request",
+    lambda: {
+      memory: 512,
+      timeout: sendPatientRequestLambdaTimeout,
+    },
+    queue: {
+      alarmMaxAgeOfOldestMessage,
+      maxMessageCountAlarmThreshold: 15_000,
+      maxReceiveCount: 3,
+      visibilityTimeout: Duration.seconds(sendPatientRequestLambdaTimeout.toSeconds() * 2 + 1),
```

Comment:
_I think_ this default formula we use for visibilityTimeout is based on maxReceiveCount = 1, but can validate here.

Can see more about it on the TSDoc of `DEFAULT_VISIBILITY_TIMEOUT_MULTIPLIER`.